lib/rstuf/client.rb

```diff
@@ -0,0 +1,37 @@
+class Rstuf::Client
+  Error = Class.new(StandardError)
+
+  def self.post_artifacts(targets)
+    response = connection.post("/api/v1/artifacts/", { targets: targets })
+
+    return response.body.dig("data", "task_id") if response.success?
+    raise Error, "Error posting artifacts: #{response.body}"
+  end
+
+  def self.delete_artifacts(targets)
+    response = connection.post("/api/v1/artifacts/delete", { targets: targets }, {})
+
+    return response.body.dig("data", "task_id") if response.success?
+    raise Error, "Error deleting artifacts: #{response.body}"
+  end
+
+  def self.task_status(task_id)
+    result = get_task(task_id)
+    result.dig("data", "state")
+  end
+
+  def self.connection
+    Faraday.new(url: Rstuf.base_url) do |f|
+      f.request :json
+      f.response :json
+      f.response :logger, Rails.logger
```

Comment:
```suggestion
      f.response :logger, logger
```

and add `include SemanticLogger::Loggable` at the top of the class